### PR TITLE
Fix unauthorized access

### LINKED ISSUE #89: Fix unauthorized access
The plugins currently allow `execTransaction` to be called from any address, not just the 4337 entry point.

(Perhaps this should be fixed by another plugin or some other mechanism?)

This is demonstrated by the (currently) failing test added in #88:

```ts
  itif("should not allow execTransaction from unrelated address", async () => {
    const { accountAddress, userWallet, provider } = await setupDeployedAccount(
      ethers.ZeroAddress,
      0,
      "0x",
    );

    const unrelatedWallet = ethers.Wallet.createRandom(provider);

    await receiptOf(
      userWallet.sendTransaction({
        to: unrelatedWallet.address,
        value: 100n * oneEther,
      }),
    );

    const account = SafeECDSAPlugin__factory.connect(
      accountAddress,
      unrelatedWallet, // <-------------------------------------- simply connect with unrelated wallet
    );

    const recipient = ethers.Wallet.createRandom(provider);

    await expect(
      receiptOf(account.execTransaction(recipient.address, oneEther, "0x")),
      //                ~~~~~~~~~~~~~~~ <------------------------ and call execTransaction
    ).to.eventually.rejected; // Should be rejected, but it isn't

    await expect(provider.getBalance(recipient)).to.eventually.equal(0n);
  });
```

### PR DESCRIPTION
**Pull request by voltrevo**
Resolves #89 by using `HandlerContext` to access `_msgSender()` and check that it's `this` or the entry point.

### CODE REVIEW

File: account-integrations/safe/src/SafeBlsPlugin.sol [hunk 1]
```diff
@@ -50,7 +52,7 @@ contract SafeBlsPlugin is BaseAccount {
         address to,
         uint256 value,
         bytes calldata data
-    ) external payable {
+    ) external payable fromThisOrEntryPoint {
         address payable safeAddress = payable(msg.sender);
```
Comment (jzaki) [inline account-integrations/safe/src/SafeBlsPlugin.sol:56]: Is `msg.sender` always going to be the safe?

File: account-integrations/safe/src/SafeBlsPlugin.sol [hunk 2]
```diff
@@ -124,4 +126,12 @@ contract SafeBlsPlugin is BaseAccount {
...
         }
     }
+
+    modifier fromThisOrEntryPoint() {
+        require(
+            _msgSender() == address(this) ||
+            _msgSender() == _entryPoint
+        );
+        _;
+    }
```
[the remark this answers is not in the document]
Comment (voltrevo) [reply account-integrations/safe/src/SafeBlsPlugin.sol:136]: Yeah probably. I'd suggest we make something like `Safe4337Base` we can inherit from, which uses `BaseAccount` and `HandlerContext` and implement this override.

[I'm wary of inheritance in general though. I've always avoided it.](https://en.wikipedia.org/wiki/Composition_over_inheritance) However, it seems popular in the smart contracts we're working with, so I'm inclined to lean into it for now. Wdyt?

File: account-integrations/safe/src/SafeBlsPlugin.sol [hunk 3]
```diff
@@ -50,7 +52,7 @@ contract SafeBlsPlugin is BaseAccount {
         address to,
         uint256 value,
         bytes calldata data
-    ) external payable {
+    ) external payable fromThisOrEntryPoint {
```
[the remark this answers is not in the document]
Comment (voltrevo) [reply account-integrations/safe/src/SafeBlsPlugin.sol:55]: Good point. I think I was just a bit hazy on that when I originally wrote this code. Will update.

File: account-integrations/safe/src/SafeBlsPlugin.sol (hunk 1 above)
Comment (voltrevo) [reply account-integrations/safe/src/SafeBlsPlugin.sol:56]: Hmm I'm not sure. I think yes *if* the function is always executed via safe's module system. The `msg.sender` here is not part of the changes in this PR, but it looks wrong to me too.

I think the original source of using `msg.sender` here is [safe's 4337 example](https://github.com/safe-global/safe-contracts/blob/810fad9/contracts/test/Test4337ModuleAndHandler.sol#L53). I opened a PR over there to change it to use `this`, curious to see what they say: https://github.com/safe-global/safe-contracts/pull/682.

**Pull request merged**